lib/dry/configurable.rb

```diff
@@ -94,11 +94,15 @@ def setting(key, *args, &block)
         end
       end
 
+      value, preprocessor = extract_preprocessor(value)
+
       _settings << ::Dry::Configurable::Config::Value.new(
         key,
         value || ::Dry::Configurable::Config::Value::NONE,
-        processor || ::Dry::Configurable::Config::DEFAULT_PROCESSOR
+        processor || ::Dry::Configurable::Config::DEFAULT_PROCESSOR,
+        preprocessor
```

Comment:
I don't know if I really like the idea having a `preprocessor`, but for the example it makes is use. So please feel free to comment as much as need it